### PR TITLE
Fix bug in `Autocomplete` component where the original `items` were changed

### LINKED ISSUE #1622: Bug in `allowCreate` in `Autocomplete`
### Which package has an issue?

@yamada-ui/react

### Package Version

1.3.10

### Description

If you put in `items` containing groups, when you create an element, the original list is also changed.

```tsx
{
  const items: AutocompleteItem[] = [
    { label: "ベジータ", value: "ベジータ" },
    {
      label: "地球人",
      items: [
        { label: "孫悟空", value: "孫悟空" },
        { label: "孫悟飯", value: "孫悟飯" },
        { label: "クリリン", value: "クリリン" },
      ],
    },
    {
      label: "フリーザ軍",
      items: [
        { label: "フリーザ", value: "フリーザ" },
        { label: "ギニュー", value: "ギニュー" },
        { label: "リクーム", value: "リクーム" },
        { label: "バータ", value: "バータ" },
        { label: "ジース", value: "ジース" },
        { label: "グルド", value: "グルド" },
      ],
    },
  ]

  return (
    <>
      <Text>allow create</Text>
      <Autocomplete
        items={items}
        allowCreate
        insertPositionItem={["地球人", "first"]}
      />
      <Text>disallow create</Text>
      <Autocomplete items={items} />
    </>
  )
}
```
https://github.com/yamada-ui/yamada-ui/assets/9676316/7efaefe0-e868-4a4a-b35f-6f208ebaa11e

### Link to Reproduction

_No response_

### Steps to Reproduce

1. Define two `Autocomplete` and `items` containing groups.
2. Pass its `items` to both `Autocomplete`.
3. Set `Autocomplete` on one side to `allowCreate` and create the element.
4. You can see that it has been added to the other `Autocomplete` element.

### Browser

None

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux

### Do you know how to fix the issue?

None

### Are you willing to participate in fixing this issue and create a pull request with the fix?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by 108yen**
Closes #1622

## Description

Fix bug where original `items` were changed.

## Current behavior (updates)

If you put in `items` containing groups, when you create an element, the original list is also changed.

## New behavior

Creates a deep copy of `items` and modifies it, so the original `items` remains unchanged.

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

**Review by suzukisan22 — COMMENTED** [review]
please fix it!

File: packages/components/autocomplete/tests/autocomplete.test.tsx [hunk 1]
```diff
@@ -688,5 +686,71 @@ describe("<Autocomplete />", () => {
...
+              label: "option2",
+              value: "option2",
+            },
+          ],
+        },
+        {
+          label: "option3",
+          value: "option3",
+        },
+      ]
```
[the remark this answers is not in the document]
Comment (108yen) [reply packages/components/autocomplete/tests/autocomplete.test.tsx]: Fixed.

File: packages/components/autocomplete/tests/autocomplete.test.tsx [hunk 2]
```diff
@@ -688,5 +686,71 @@ describe("<Autocomplete />", () => {
...
+      await user.click(autocomplete!)
+
+      const input = screen.getByRole("combobox")
+      await user.type(input, CREATE_OPTION_VALUE)
+      await user.keyboard("{Enter>}")
+
+      await waitFor(() => {
+        const optionElements = screen.getAllByRole(AUTOCOMPLETE_ITEM_ROLE)
+        expect(optionElements[1]).toHaveTextContent(CREATE_OPTION_VALUE)
+      })
```
[the remark this answers is not in the document]
Comment (108yen) [reply packages/components/autocomplete/tests/autocomplete.test.tsx]: Fixed.

**Comment by 108yen** [comment]
@suzukisan22 
Thanks for the review! All have been corrected!

**Review by suzukisan22 — APPROVED** [review]
👍

**Pull request merged**